components/admin_console/admin_navbar_dropdown/admin_navbar_dropdown.jsx

```diff
@@ -86,139 +33,60 @@ export default class AdminNavbarDropdown extends React.Component {
 
             for (const team of teamsArray) {
                 teamToRender.push(
-                    <li key={'team_' + team.name}>
-                        <BlockableLink
-                            id={'swithTo' + Utils.createSafeId(team.name)}
-                            to={'/' + team.name + `/channels/${Constants.DEFAULT_CHANNEL}`}
-                        >
-                            <FormattedMessage
-                                id='navbar_dropdown.switchTo'
-                                defaultMessage='Switch to '
-                            />
-                            {team.display_name}
-                        </BlockableLink>
-                    </li>
+                    <MenuItemBlockableLink
+                        key={'team_' + team.name}
+                        to={'/' + team.name + `/channels/${Constants.DEFAULT_CHANNEL}`}
+                        text={Utils.localizeMessage('navbar_dropdown.switchTo', 'Switch to ') + ' ' + team.display_name}
```

Comment:
What if instead of the parent component doing the localization, let the child component itself (e.g. MenuItemBlockableLink, MenuItemAction, etc) handles it?  
Instead of text, it could pass the translation object like `{id: "navbar_dropdown.switchTo", defaultMessage: "Switch to {displayName}", values: team.display_name}`.

I see an exception where a component is used by the plugin like the `MenuItemAction` where `item.text` is being passed into it.  But maybe an optional rendering in that case will work.

Reply:
I get your point here, but I don't have an strong opinion. Probably other approach is to use textId, textDefault and textValues properties, which is more inline with other components.

Reply:
@grundleborg any strong opinion here?

Reply:
Feel like we have a generic problem to solve with these components around how we put i18nable strings into them. I'm going to throw some notes together on the different patterns we have in the wild/have tried using and then maybe we can discuss pros/cons and see if we can come up with 1 solution to make part of the Widget API Style Guide.